### ISSUE TITLE
Cascade of errors in my current configuration, very confusing.

### ISSUE THREAD

**Issue by threeheadeddingo** [ValveSoftware/steam-audio#255]
Currently using:

Unity 2022.2.14f1
FMOD for Unity/Studio 2.02.13
Steam Audio Unity 4.14

No matter what I do, Steam Audio just won't play nice. I get a cascade of errors about DSPs not being found, FMOD events not being found, etc. and when I remove Steam Audio, everything works beautifully again. When trying to do a Steam Audio > Install FMOD plugins, more errors.

I had to manually add the phonon plugins to my FMOD project to get the spatializer to show up in my FMOD project. I can add them to events just fine. But again, getting any audio to play with this combo of Unity/FMOD/Steam Audio versions is just exploding in my face. So many errors I'm not even sure where to begin.

Can anyone verify that this configuration is just broken, to rule out that I'm doing something wrong? To be clear, I've used Steam Audio in the past with other versions of Unity and everything has been fine. So this isn't my first time.

**Comment by lakulish** [comment]
Can you provide more details as to what error messages you're seeing? And just to be sure, can you double-check that you've followed the instructions for setting up Steam Audio in your Unity and FMOD Studio projects:

1. Add Steam Audio to your Unity project: [docs](https://valvesoftware.github.io/steam-audio/doc/unity/getting-started.html#add-steam-audio-to-your-unity-project)
2. Add Steam Audio to your FMOD Studio project: [docs](https://valvesoftware.github.io/steam-audio/doc/fmod/getting-started.html#add-steam-audio-to-your-fmod-studio-project)
3. Configure FMOD Studio to load the Steam Audio plugin in Unity, and configure Steam Audio to use FMOD Studio as the audio engine: [docs](https://valvesoftware.github.io/steam-audio/doc/fmod/getting-started.html#integrate-steam-audio-with-your-game-engine)

Let me know if following the above instructions with blank Unity and FMOD Studio projects leads to errors.

**Comment by threeheadeddingo** [comment]
So I've done some more deep diving since I posted this (I apologize for the initial post being a bit vague, I was quite tired).

I think this could have to do with Steam Audio not playing well with the latest version of FMOD. This time I created a perfectly clean Unity project, this time in the latest 2021 LTS (to rule that out as an issue, that I was using a tech version).

I added FMOD, Steam Audio, and went through the proper configuration to get the project linked up.

First problem I noticed was that I had to manually move plugin files. Using Steam Audio 4.1.4, it seems it cannot perform the Steam Audio > Install FMOD Studio Plugin Files operation on its own, and throws an error in Unity.

Secondly, when I add an empty GameObject to the scene, add FMOD Event Emitter and a Steam Audio Source components to it, and go into play mode, I get a bunch of errors. I CAN hear the audio properly and it even seems to properly be spatializing (I've added the proper spatializer to the event in FMOD, etc.), however these errors do occur which is still troubling:

```
[FMOD] EventInstance::getChannelGroup(000001B169864730) returned ERR_STUDIO_NOT_LOADED for STUDIO_EVENTINSTANCE (0x200B60).
UnityEngine.Debug:LogError (object)
FMODUnity.RuntimeUtils:DebugLogError (string) (at Assets/Plugins/FMOD/src/RuntimeUtils.cs:585)
FMODUnity.RuntimeManager:ERROR_CALLBACK (intptr,FMOD.SYSTEM_CALLBACK_TYPE,intptr,intptr,intptr) (at Assets/Plugins/FMOD/src/RuntimeManager.cs:108)
FMOD.Studio.EventInstance:getChannelGroup (FMOD.ChannelGroup&) (at Assets/Plugins/FMOD/src/fmod_studio.cs:1393)
System.Reflection.MethodBase:Invoke (object,object[])
SteamAudio.FMODStudioAudioEngineSource:FindDSP (UnityEngine.GameObject) (at Assets/Plugins/SteamAudio/Scripts/Runtime/FMODStudioAudioEngineSource.cs:99)
SteamAudio.FMODStudioAudioEngineSource:Initialize (UnityEngine.GameObject) (at Assets/Plugins/SteamAudio/Scripts/Runtime/FMODStudioAudioEngineSource.cs:36)
SteamAudio.SteamAudioSource:Awake () (at Assets/Plugins/SteamAudio/Scripts/Runtime/SteamAudioSource.cs:163)
```

I actually get several of these. One from line 163 (above), and from lines 164, 221, 193, 235.

Then, once I end play mode, I get 2 last errors.

```
UnityEngine.Debug:LogError (object)
FMODUnity.RuntimeUtils:DebugLogError (string) (at Assets/Plugins/FMOD/src/RuntimeUtils.cs:585)
FMODUnity.RuntimeManager:DEBUG_CALLBACK (FMOD.DEBUG_FLAGS,intptr,int,intptr,intptr) (at Assets/Plugins/FMOD/src/RuntimeManager.cs:77)
```

```
[FMOD] AsyncManager::asyncThreadLoop : System::update returned error 28.

UnityEngine.Debug:LogError (object)
FMODUnity.RuntimeUtils:DebugLogError (string) (at Assets/Plugins/FMOD/src/RuntimeUtils.cs:585)
FMODUnity.RuntimeManager:DEBUG_CALLBACK (FMOD.DEBUG_FLAGS,intptr,int,intptr,intptr) (at Assets/Plugins/FMOD/src/RuntimeManager.cs:77)
```

It's worth noting that all of these errors go away if I simply remove Steam Audio from my project and remove its spatializers from my events. This is why I strongly suspect this is a Steam Audio issue. I'm not sure, of course.

**Comment by lakulish** [comment]
Okay, I am able to reproduce the issue when using FMOD Studio 2.02.13. At some point on or after FMOD Studio 2.02.08, the `FMODUnity.Settings.CurrentVersion` field was changed from `public` to `internal`, which causes the **Steam Audio** > **Install FMOD Studio Plugin Files** command to fail in Unity with a `NullPointerException`.

Because the Steam Audio Unity plugin contains the FMOD Studio plugin files in the older location (compatible with 2.01 and earlier), this leads to all the errors you're encountering. As a workaround, you can manually do what the **Install FMOD Studio Plugin Files** menu command does: for example, for Windows 64-bit, move `phonon_fmod.dll` from `Assets/Plugins/FMOD/lib/win/x86_64` to `Assets/Plugins/FMOD/platforms/win/lib/x86_64`. Can you try this and let me know if it fixes the issue?

(The C# code for `SteamAudioManager.InstallFMODStudioPluginFiles` contains the full list of files and the locations to which they need to be moved.)

**Comment by lakulish** [comment]
We have just released v4.2.0 (https://github.com/ValveSoftware/steam-audio/releases/tag/v4.2.0) which moves the FMOD plugin files to the correct place for FMOD Studio 2.2+ by default, so this should hopefully no longer be an issue.
